test/test_decoders.py

```diff
@@ -1737,6 +1743,69 @@ def test_set_cuda_backend(self):
                 with set_cuda_backend(backend):
                     VideoDecoder(H265_VIDEO.path, device=f"cuda:{bad_device_number}")
 
+    def test_cpu_fallback_no_fallback_on_cpu_device(self):
+        """Test that CPU device doesn't trigger fallback (it's not a fallback scenario)."""
+        decoder = VideoDecoder(NASA_VIDEO.path, device="cpu")
+
+        _ = decoder[0]
+
+        assert decoder.cpu_fallback.status_known
+        assert not bool(decoder.cpu_fallback)
+        assert "No fallback required" in str(decoder.cpu_fallback)
+
+    @needs_cuda
+    def test_cpu_fallback_h265_video_ffmpeg_cuda(self):
+        """Test that H265 video triggers CPU fallback on FFmpeg CUDA interface."""
+        # H265_VIDEO is known to trigger CPU fallback on FFmpeg CUDA
+        # because its dimensions are too small
+        decoder = VideoDecoder(H265_VIDEO.path, device="cuda")
+
+        assert not decoder.cpu_fallback.status_known
+
+        _ = decoder.get_frame_at(0)
+
+        assert decoder.cpu_fallback.status_known
+        assert bool(decoder.cpu_fallback)
+        assert "Fallback status: Falling back due to:" in str(decoder.cpu_fallback)
+
+    @needs_cuda
+    def test_cpu_fallback_no_fallback_on_supported_video(self):
+        """Test that supported videos don't trigger fallback on CUDA."""
+        decoder = VideoDecoder(NASA_VIDEO.path, device="cuda")
+
+        _ = decoder[0]
+
+        assert not bool(decoder.cpu_fallback)
+        assert "No fallback required" in str(decoder.cpu_fallback)
+
+    def test_cpu_fallback_status_cached(self):
+        """Test that cpu_fallback status is determined once and then cached."""
+        decoder = VideoDecoder(NASA_VIDEO.path)
+
+        _ = decoder[0]
+        first_status = str(decoder.cpu_fallback)
+        assert decoder.cpu_fallback.status_known
+
+        _ = decoder[1]
+        second_status = str(decoder.cpu_fallback)
+        assert decoder.cpu_fallback.status_known
+
+        assert first_status == second_status
+
+    def test_cpu_fallback_multiple_access_methods(self):
```

Comment:
I think this test is technically a subset of the previous one: if we 'cache' the `cpu_fallback` result, then a consequence is that calling different methods isn't going to change it.

Maybe want you wanted to check is that the status becomes "known" and that it works with multiple decoding method? If that's the case I think we'd need to re-create the VideoDecoder object in-between method calls. But TBH, I'm not sure it's a critical test to have, so I might suggest to remove it too.